src/elements/content-explorer/GridViewItem.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+import * as React from 'react';
+import getSize from '../../utils/size';
+import moreOptionsCellRenderer from './moreOptionsCellRenderer';
+import dateCellRenderer from './dateCellRenderer';
+import nameCellRenderer from '../common/item/nameCellRenderer';
+import { getIcon } from '../common/item/iconCellRenderer';
+import './GridViewItem.scss';
+
+type Props = {
+    canDelete: boolean,
+    canDownload: boolean,
+    canPreview: boolean,
+    canRename: boolean,
+    canShare: boolean,
+    isSmall: boolean,
+    isTouch: boolean,
+    item: ?BoxItem,
+    onItemClick: (item: BoxItem | string) => void,
+    onItemDelete: (item: BoxItem) => void,
+    onItemDownload: (item: BoxItem) => void,
+    onItemPreview: (item: BoxItem) => void,
+    onItemRename: (item: BoxItem) => void,
+    onItemSelect: (item: BoxItem, callback: Function) => void,
+    onItemShare: (item: BoxItem) => void,
+    rootFolderId: string,
+    view: View,
+};
+
+const GridViewItem = ({
+    canDelete,
+    canDownload,
+    canPreview,
+    canRename,
+    canShare,
+    isSmall,
+    isTouch,
+    item,
+    onItemClick,
+    onItemDelete,
+    onItemDownload,
+    onItemPreview,
+    onItemRename,
+    onItemSelect,
+    onItemShare,
+    rootFolderId,
+    view,
+}: Props) => {
+    if (!item) {
+        return <div />;
+    }
+
+    const moreOptionsCell = moreOptionsCellRenderer(
+        canPreview,
+        canShare,
+        canDownload,
+        canDelete,
+        canRename,
+        onItemSelect,
+        onItemDelete,
+        onItemDownload,
+        onItemRename,
+        onItemShare,
+        onItemPreview,
+        isSmall,
+    );
+
+    const nameCell = nameCellRenderer(
+        rootFolderId,
+        view,
+        onItemClick,
+        onItemSelect,
+        canPreview,
+        isSmall, // shows details if false
+        isTouch,
+    );
+
+    const dateCell = dateCellRenderer();
+
+    return (
+        <div>
+            <div className="bdl-GridViewItem-itemThumbnail">
+                <div className="bdl-GridViewItem-itemIcon"> {getIcon(128, item)} </div>
+            </div>
+            <div>
+                {nameCell({ rowData: item })}
```

Comment:
Can these be invoked as regular components rather than as functions, like `<NameCell />`?